File: CS/TestMyButtonEdit/MyButtonEdit.cs
using System;
using System.ComponentModel;
using DevExpress.XtraEditors;




namespace DevExpress.MyControl
{
    public class MyButtonEdit : ButtonEdit
    {
        public override string EditorTypeName
        {
            get
            {
                return RepositoryItemMyButtonEdit.EDITORTypeName;
            }
        }



        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemMyButtonEdit Properties
        {
            get
            {
                return base.Properties as RepositoryItemMyButtonEdit;
            }
        }



        static MyButtonEdit()
        {
            RepositoryItemMyButtonEdit.RegisterMyButtonEdit();
        }



        public MyButtonEdit()
            : base()
        {
        }


        protected override void OnEditValueChanged()
        {
            base.OnEditValueChanged();

            if (Properties.Buttons.Count >= 2)
            {
                string str_value = EditValue as string;
                Properties.Buttons[1].Caption = (str_value == null ? string.Empty : str_value);
            }
        }
    }
}

File: VB/TestMyButtonEdit/MyButtonEdit.vb
Imports System
Imports System.ComponentModel
Imports DevExpress.XtraEditors




Namespace DevExpress.MyControl
    Public Class MyButtonEdit
        Inherits ButtonEdit

        Public Overrides ReadOnly Property EditorTypeName() As String
            Get
                Return RepositoryItemMyButtonEdit.EDITORTypeName_Renamed
            End Get
        End Property



        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public Shadows ReadOnly Property Properties() As RepositoryItemMyButtonEdit
            Get
                Return TryCast(MyBase.Properties, RepositoryItemMyButtonEdit)
            End Get
        End Property



        Shared Sub New()
            RepositoryItemMyButtonEdit.RegisterMyButtonEdit()
        End Sub



        Public Sub New()
            MyBase.New()
        End Sub


        Protected Overrides Sub OnEditValueChanged()
            MyBase.OnEditValueChanged()

            If Properties.Buttons.Count >= 2 Then
                Dim str_value As String = TryCast(EditValue, String)
                Properties.Buttons(1).Caption = (If(str_value Is Nothing, String.Empty, str_value))
            End If
        End Sub
    End Class
End Namespace





